File: Csharp/sigmf/sigmf/examples/location/ProjectedExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.location
{
    public class ProjectedExample : Example
    {
        public override MetaDoc getExample()
        {
            DateTime now = getCurrentTime();
            MetaDoc metaDoc = new MetaDoc();

            Globals globals = ExampleUtils.getGlobals(now);
            List<Capture> captures = ExampleUtils.getCaptures(now);
            metaDoc.setGlobal(globals);
            metaDoc.setCaptures(captures);
            GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
            gcs.Id = "EPSG:4608";
            gcs.Description = "NAD27(76)";
            ProjectedCoordnateSystem projectedCoordnateSystem = new ProjectedCoordnateSystem();
            projectedCoordnateSystem.Id = "EPSG:2029";
            projectedCoordnateSystem.Description = "NAD27(76) / UTM zone 17N";
            projectedCoordnateSystem.Gcs = gcs;
            globals.CoordinateSystem = projectedCoordnateSystem;
            return metaDoc;

        }

        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}

File: java/src/main/java/gov/doc/ntia/sigmf/examples/location/ProjectedExample.java
package gov.doc.ntia.sigmf.examples.location;

import gov.doc.ntia.sigmf.Capture;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.global.location.GeographicCoordinateSystem;
import gov.doc.ntia.sigmf.ext.global.location.ProjectedCoordnateSystem;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class ProjectedExample implements Example {
  @Override
  public MetaDoc getExample() {
    Date now = Calendar.getInstance().getTime();
    MetaDoc metadoc = new MetaDoc();

    Global global = ExampleUtils.getGlobal();
    List<Capture> captures = ExampleUtils.getCaptures(now);
    metadoc.setGlobal(global);
    metadoc.setCaptures(captures);
    GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
    gcs.setId("EPSG:4608");
    gcs.setDescription("NAD27(76)");
    ProjectedCoordnateSystem projectedCoordnateSystem = new ProjectedCoordnateSystem();
    projectedCoordnateSystem.setId("EPSG:2029");
    projectedCoordnateSystem.setDescription("NAD27(76) / UTM zone 17N");
    projectedCoordnateSystem.setGcs(gcs);
    global.setCoordinateSystem(projectedCoordnateSystem);
    return metadoc;
  }
}
